### PR TITLE
[1LP][RFR] Fixed timing issues for Openstack test_networks tests

### PR DESCRIPTION
**Pull request by dkholodo**
Added wait_for statements where it needed

{{ pytest: cfme/tests/openstack/cloud/test_networks.py::test_delete_network cfme/tests/openstack/cloud/test_networks.py::test_edit_subnet cfme/tests/openstack/cloud/test_networks.py::test_delete_subnet }}

### CODE REVIEW

File: cfme/tests/openstack/cloud/test_networks.py [hunk 1]
```diff
@@ -131,6 +133,9 @@ def test_delete_network(network):
     network.delete()
     wait_for(network.provider_obj.is_refreshed, func_kwargs=dict(refresh_delta=10), timeout=600,
              delay=10)
+    network.browser.refresh()
+    wait_for(lambda: network.exists is False,
```
[the remark this answers is not in the document]
Comment (dkholodo) [reply cfme/tests/openstack/cloud/test_networks.py]: fixed

File: cfme/tests/openstack/cloud/test_networks.py [hunk 2]
```diff
@@ -158,6 +165,8 @@ def test_delete_subnet(subnet):
     wait_for(subnet.provider_obj.is_refreshed, func_kwargs=dict(refresh_delta=10), timeout=600,
              delay=10)
     subnet.browser.refresh()
+    wait_for(lambda: subnet.exists is False,
```
[the remark this answers is not in the document]
Comment (dkholodo) [reply cfme/tests/openstack/cloud/test_networks.py]: fixed

**Comment by RonnyPfannschmidt** [comment]
i just noted, that the metadata declaration is incorrect - its not possible to have more than one {{ ...}} block for the metadata

please use a single declaration to select everything

in case you took this example from somewhere else please reffer the source so we can get to the root of this

**Comment by dkholodo** [comment]
Fixed.
Not it is used a single declaration to select everything.

File: cfme/tests/openstack/cloud/test_networks.py [hunk 3]
```diff
@@ -123,6 +123,8 @@ def test_edit_network(network):
     wait_for(network.provider_obj.is_refreshed, func_kwargs=dict(refresh_delta=10), timeout=600,
              delay=10)
     network.browser.refresh()
```
Comment (digitronik) [inline cfme/tests/openstack/cloud/test_networks.py]: Required:
No need of this as you are using `wait_for` and refreshing `browser`

File: cfme/tests/openstack/cloud/test_networks.py [hunk 4]
```diff
@@ -131,6 +133,9 @@ def test_delete_network(network):
     network.delete()
     wait_for(network.provider_obj.is_refreshed, func_kwargs=dict(refresh_delta=10), timeout=600,
              delay=10)
+    network.browser.refresh()
```
Comment (digitronik) [inline cfme/tests/openstack/cloud/test_networks.py]: Same as above

File: cfme/tests/openstack/cloud/test_networks.py (hunk 3 above)
Comment (dkholodo) [reply cfme/tests/openstack/cloud/test_networks.py]: Thanks. I agree that I should delete refreshing browser. But without this `wait_for` this test is failed.

File: cfme/tests/openstack/cloud/test_networks.py (hunk 4 above)
Comment (dkholodo) [reply cfme/tests/openstack/cloud/test_networks.py]: Same as above.

**Comment by dajoRH** [comment]
Lint report for commit 6438154db02de46b5ce42bf1180eedd22178a3c0:

`cfme/tests/openstack/cloud/test_networks.py`:
- :heavy_exclamation_mark: Line 248:39: **W292** *no newline at end of file*

Please, rectify these issues :smirk: .

* 89bf81a0fd83bff98d83eb0938190939655a27f4
  * Merge commit detected!!! :bangbang:
* 6438154db02de46b5ce42bf1180eedd22178a3c0
  * Merge commit detected!!! :bangbang:

**The PR might not be acceptable in this state** :rage4:.
There is a merge commit in the branch which we do not like in PRs so please, get rid of it.

*CFME QE Bot*

File: cfme/tests/openstack/cloud/test_networks.py (hunk 3 above)
Comment (digitronik) [reply cfme/tests/openstack/cloud/test_networks.py]: I am did not talk about removing `wait_for`

**Comment by dajoRH** [comment]
Lint report for commit 057546a355db963addc1fb248e1edd787224400c:

Lint OK! :cake: :punch: :cookie: .

* 89bf81a0fd83bff98d83eb0938190939655a27f4
  * Merge commit detected!!! :bangbang:
* d565af7874ba1114c3a9090f758f7f92c988971a
  * Merge commit detected!!! :bangbang:
* 057546a355db963addc1fb248e1edd787224400c
  * Merge commit detected!!! :bangbang:

**The PR might not be acceptable in this state** :rage4:.
There is a merge commit in the branch which we do not like in PRs so please, get rid of it.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 4a8db1a9b363010e7ff8a69fcedd119943b944a9:

Lint OK! :cake: :punch: :cookie: .

* 89bf81a0fd83bff98d83eb0938190939655a27f4
  * Merge commit detected!!! :bangbang:
* d565af7874ba1114c3a9090f758f7f92c988971a
  * Merge commit detected!!! :bangbang:
* 057546a355db963addc1fb248e1edd787224400c
  * Merge commit detected!!! :bangbang:
* 07bf2d08c2d9e0d04698b284819f0dcaa78227cb
  * Merge commit detected!!! :bangbang:
* 4a8db1a9b363010e7ff8a69fcedd119943b944a9
  * Merge commit detected!!! :bangbang:

**The PR might not be acceptable in this state** :rage4:.
There is a merge commit in the branch which we do not like in PRs so please, get rid of it.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit ffcd649897ee99f884a466266fea289b3338fdd4:

`cfme/tests/openstack/cloud/test_instances.py`:
- :heavy_exclamation_mark: Line 226:59: **W292** *no newline at end of file*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/openstack/cloud/test_networks.py (hunk 3 above)
Comment (dkholodo) [reply cfme/tests/openstack/cloud/test_networks.py]: fixed

File: cfme/tests/openstack/cloud/test_networks.py (hunk 4 above)
Comment (dkholodo) [reply cfme/tests/openstack/cloud/test_networks.py]: fixed

**Comment by digitronik** [comment]
@dkholodo I still found some irrelevant `browser.refesh()`; can you please try to remove those? If we have `wait_for` with `browser refresh` we won't need `refresh` before it. `wait_for` will take care of it. 
L155
L165
L189
L224
I know some are not changed by you... but still, we can improve this...

**Comment by dkholodo** [comment]
> @dkholodo I still found some irrelevant `browser.refesh()`; can you please try to remove those? If we have `wait_for` with `browser refresh` we won't need `refresh` before it. `wait_for` will take care of it.
> L155
> L165
> L189
> L224
> I know some are not changed by you... but still, we can improve this...

Fixed.
Thanks for your comment.

**Comment by dajoRH** [comment]
Lint report for commit cf80b41b52e8a51c1fb730c623158a787164eceb:

`cfme/tests/openstack/cloud/test_networks.py`:
- :bangbang: Line 235:101: **E501** *line too long (113 > 100 characters)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/openstack/cloud/test_networks.py [hunk 5]
```diff
@@ -208,20 +218,24 @@ def test_add_gateway_to_router(router, ext_subnet):
...
+    wait_for(router.provider_obj.is_refreshed, func_kwargs=dict(refresh_delta=20), timeout=800,
+             delay=30)
     # TODO: verify the exact entities' names and relationships, not only count
-    view = navigate_to(router, 'Details')
-    subnets_count = int(view.entities.relationships.get_text_of('Cloud Subnets'))
-    assert subnets_count == 1  # Compare to '1' because clean router was used initially
+    wait_for(lambda: int(view.entities.relationships.get_text_of('Cloud Subnets')) ==
+             (subnets_count_before_adding + 1),
+             delay=30, timeout=800, fail_func=router.browser.refresh)
+    subnets_count_after_adding = int(view.entities.relationships.get_text_of('Cloud Subnets'))
```
Comment (digitronik) [inline cfme/tests/openstack/cloud/test_networks.py]: Required:
@dkholodo: No need to capture `subnets_count_after_adding` after `wait_for`. If `wait_for` fails just capture the exception and `assert` `False`... like
```
try:
    wait_for(
        lambda: int(view.entities.relationships.get_text_of("Cloud Subnets"))
        == (subnets_count_before_adding + 1),
        delay=30,
        timeout=800,
        fail_func=router.browser.refresh,
    )
except TimedOutError:
    assert False, "Some expaination about failuers"
```

Comment (dkholodo) [reply cfme/tests/openstack/cloud/test_networks.py]: @digitronik Fixed

**Comment by digitronik** [comment]
@dkholodo I restarted PRT run... 5.10 is failing while creating `subnet` for `test_delete_subnet`.

**Comment by dkholodo** [comment]
> I restarted PRT run... 5.10 is failing while creating `subnet` for `test_delete_subnet`

@digitronik I've checked with our jobs for 5.10 and 5.9. `subnet` is created successfully for `test_delete_subnet`

**Comment by dkholodo** [comment]
> I restarted PRT run... 5.10 is failing while creating `subnet` for `test_delete_subnet`

@digitronik I've checked with our jobs for 5.10 and 5.9. `subnet` is created successfully for `test_delete_subnet`

**Pull request merged**